libs/designer/src/lib/ui/templates/connections/workflowconnections.tsx

```diff
@@ -183,10 +186,12 @@ export const WorkflowConnections = ({ connections }: WorkflowConnectionsProps) =
   );
 
   const onConnectionsLoaded = async (loadedConnections: Connection[], item: ConnectionItem): Promise<void> => {
+    const connectionIdToOverride = viewTemplateDetails?.connectionsOverride?.[item.connectionKey]?.connectionId;
+    const connectionToOverride = loadedConnections.find((connection) => connection.id === connectionIdToOverride);
     const itemHasConnection = item.connection?.id && item.connection?.displayName === undefined;
-    const connectionToUse = itemHasConnection
-      ? loadedConnections.find((connection) => connection.id === item.connection?.id)
-      : loadedConnections[0];
+    const connectionToUse =
+      connectionToOverride ??
+      (itemHasConnection ? loadedConnections.find((connection) => connection.id === item.connection?.id) : loadedConnections[0]);
```

Comment:
YOu should be using the connectionToOverride in the else portion

itemhasConnection ? honor the selected connection : [overridenConnection ?? connections[0]]

Reply:
I think connectionsOverride should overrides the existing connections, that's the purpose of introducing this property, right?

Reply:
you are just using connectionsOverride to select a connection from the list, so when first time when you load it will auto select from connectionsOverride, but now when teh user intentionally changes from teh dropdown - switch tab and comeback you wont be honoring that selection